### ISSUE TITLE
Sort entries by popularity and-or activity

### ISSUE THREAD

**Issue by funkyfisch** [awesome-selfhosted/awesome-selfhosted#2534]
A feature proposal I have, is to have all entries in each category ordered by popularity (calculated by stars e.g.) or by activity. Maybe the average commits per month for the last 6 months can be calculated to be used as an activity index.

Additionally, the last time a project was updated could also be added to the end of each entry so that it can be a mental indicator for people skimming through the different entries (some of them are not that active anymore even though they have many open issues)

**Comment by nodiscc** [comment]
Blocked by https://github.com/awesome-selfhosted/awesome-selfhosted/issues/1038

It is planned to move the list data to a machine-readable format, and let a generator ... generate the final documents (mutli-page HTML, single-page HTML, single-page markdown). This will add the ability to gather up-to-date metadata at each build (with the current, markdown-only list we would have to update the stars count manually... no way)

> the last time a project was updated could also be added to the end of each entry

Yes. We already have a manual check for last commit dates for Github projects, but they are not exposed in the final document (https://github.com/awesome-selfhosted/awesome-selfhosted/blob/master/tests/check-github-commit-dates.py), it is definitely planned to add this as part of the issue above.

> have all entries in each category ordered by popularity (calculated by stars e.g.) or by activity

This would only work for projects hosted on Github (and possibly Gitlab, and possibly other forges which we can detect and parse the API... but it requires some work, and will not work for all projects). So we will keep the alphabetical ordering for the first versions of the YAML-based/generated site. Stars/activity level will definitely be added ad the end of each entry, if available.

Later, we could add the ability to sort the list/table using javascript ([wikipedia/mediawiki implements this](https://en.wikipedia.org/wiki/List_of_free_and_open-source_web_applications))

**Comment by funkyfisch** [comment]
Any pointers on contributing on any of these enabling work items?
Meanwhile I can start looking into [#1038](https://github.com/awesome-selfhosted/awesome-selfhosted/issues/1038) and see if I can help in anyway

**Comment by nodiscc** [comment]
> move the list data to a machine-readable format https://github.com/awesome-selfhosted/awesome-selfhosted/issues/1038

In progress https://github.com/awesome-selfhosted/awesome-selfhosted-data

> ability to gather up-to-date metadata at each build
> the last time a project was updated could also be added to the end of each entry
> generate the final documents (single-page markdown)

Done  https://github.com/nodiscc/hecat

> multi-page HTML, single-page HTML

In progress https://github.com/nodiscc/hecat/issues/15

> have all entries in each category ordered by popularity (calculated by stars e.g.) or by activity

https://github.com/nodiscc/hecat/issues/52

**Comment by nodiscc** [comment]
Hi,
thank you for your contribution. Today an improved version of awesome-selfhosted goes live at https://awesome-selfhosted.net/. Please check the [release announcement](https://github.com/awesome-selfhosted/awesome-selfhosted/releases/tag/1.0.0) for more details.

Contributions should now be made through https://github.com/awesome-selfhosted/awesome-selfhosted-data. If you are still interested in having this issue worked on, please check contribution guidelines there, and open a new PR/issue.

Sorry for the inconvenience.
